velox/exec/RowContainer.h

```diff
@@ -817,13 +837,20 @@ class RowContainer {
   /// invalidated. Any row erase operations will invalidate column stats.
   std::optional<RowColumn::Stats> columnStats(int32_t columnIndex) const;
 
+  uint32_t columnNullCount(int32_t columnIndex) {
+    return rowColumnsStats_[columnIndex].nullCount();
+  }
+
   const auto& keyTypes() const {
     return keyTypes_;
   }
 
-  /// Returns true if specified column may have nulls, false otherwise.
+  /// Returns true if specified column has nulls, false otherwise.
+  /// if there is no data, return true to avoid problematic optimizations
+  /// based on no-nulls assumption.
   inline bool columnHasNulls(int32_t columnIndex) const {
-    return columnHasNulls_[columnIndex];
+    return columnStats(columnIndex)->numCells() == 0 ||
```

Comment:
I think we shall return false if there is no cells to be align with the current behavior